index.js

```diff
@@ -160,15 +159,23 @@ class Dynamodb extends Datastore {
         const client = this.client[config.table];
         const limitTotalCount = config.paginate.page * config.paginate.count;
         const startIndex = (config.paginate.page - 1) * config.paginate.count;
+        const filterParams = config.params;
 
         if (!client) {
             const err = new Error(`Invalid table name "${config.table}"`);
 
             return callback(err);
         }
 
-        return client.scan().limit(limitTotalCount)  // scan all items up to and including last item specified
-        .exec((err, data) => {
+        let scanner = client.scan().limit(limitTotalCount);
```

Comment:
Ugh just kidding. Spiked it out and if you have `limit()` anywhere, it will only fetch that many records from the db